inference-engine/tests/functional/inference_engine/keep_constant_inputs_tests.cpp

```diff
@@ -0,0 +1,156 @@
+#include <convert_function_to_cnn_network.hpp>
+#include <ngraph_ops/fully_connected.hpp>
+#include <transformations/common_optimizations/common_optimizations.hpp>
+#include <transformations/convert_opset1_to_legacy/convert_opset1_to_legacy.hpp>
+#include <transformations/convert_opset2_to_opset1/convert_opset2_to_opset1.hpp>
+#include <transformations/convert_opset3_to_opset2/convert_opset3_to_opset2.hpp>
+#include "generic_ie.hpp"
+#include "graph_tools.hpp"
+#include "functional_test_utils/low_precision_transformations/layer_transformation.hpp"
+
+using namespace testing;
+using namespace InferenceEngine;
+
+int numberOfInputsForLayerInCNNNetwork(const InferenceEngine::CNNNetwork& network, std::string layerType) {
+    int numberOfInputs = 0;
+    InferenceEngine::CNNLayerPtr layer = nullptr;
+
+    IE_SUPPRESS_DEPRECATED_START
+    for (auto it = details::CNNNetworkIterator(network); it != details::CNNNetworkIterator(); it++) {
+        layer = *it;
+        if (layer->type == layerType) {
+            numberOfInputs = layer->insData.size();
+            std::cout << "Found layer: " << layer->name << " of type: " << layer->type << std::endl;
+            std::cout << "Number of inputs: " << numberOfInputs << std::endl;
+            break;
+        }
+    }
+    IE_SUPPRESS_DEPRECATED_END
+    return numberOfInputs;
+}
+
+void transformNetwork(std::shared_ptr<InferenceEngine::ICNNNetwork> clonedNetwork, bool keep_constant_inputs) {
+    if (clonedNetwork->getFunction()) {
+        const auto transformations_callback = [](const std::shared_ptr<const ::ngraph::Node> &node) -> bool {
+            // DepthToSpace node implementation supports only equal input/output tensors with rank <= 5
+            if (auto dtsOp = std::dynamic_pointer_cast<const ::ngraph::opset3::DepthToSpace>(node)) {
+                return dtsOp->input_value(0).get_shape().size() <= 5lu &&
+                       dtsOp->input_value(0).get_shape().size() == dtsOp->get_output_shape(0).size();
+            }
+
+            // SpaceToDepth node implementation supports only equal input/output tensors with rank <= 5
+            if (auto stdOp = std::dynamic_pointer_cast<const ::ngraph::opset3::SpaceToDepth>(node)) {
+                return stdOp->input_value(0).get_shape().size() <= 5lu &&
+                       stdOp->input_value(0).get_shape().size() == stdOp->get_output_shape(0).size();
+            }
+
+            return std::dynamic_pointer_cast<const ::ngraph::opset2::Gelu>(node) ||
+                   std::dynamic_pointer_cast<const ::ngraph::opset2::BatchToSpace>(node) ||
+                   std::dynamic_pointer_cast<const ::ngraph::opset2::SpaceToBatch>(node) ||
+                   std::dynamic_pointer_cast<const ::ngraph::opset3::ShuffleChannels>(node);
+        };
+        auto nGraphFunc = clonedNetwork->getFunction();
+        // Disable shape inference (WA for generic operations)
+        ::ngraph::op::GenericIE::DisableReshape noReshape(nGraphFunc);
```

Comment:
you can remove this line, your tests don't create GenericIE operations.

Reply:
Done.